### PR TITLE
Fix service constructor initialization issues

### PR DESCRIPTION
**Pull request by rli**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
* JetBrains started throwing errors in the latest EAP if a service constructor takes a parameter that can't be auto-injected. They provide a new annotation `@NonInjectable`, but that is not available unless you pull in the EAP SDK, so instead fix by removing those constructors.

## Testing
Look no error
image

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/software/aws/toolkits/core/telemetry/TelemetryBatcher.kt [hunk 1]
```diff
@@ -76,6 +81,13 @@ open class DefaultTelemetryBatcher(
         flush(retry, isTelemetryEnabled.get())
     }
 
+    @Synchronized
+    override fun shutdownAndFlush(batcher: TelemetryBatcher) {
```
Comment (hunterwerlla) [inline core/src/software/aws/toolkits/core/telemetry/TelemetryBatcher.kt]: nit: this method name is weird, I first read it as we are shutting down and flushing the batcher that is passed in. Maybe something like `setBatcher` would be better? What we are actually doing is setting a new batcher in a way that works.

**Comment by hunterwerlla** [comment]
Can you add in the PR why this actually fixes it?

**Comment by justinmk3** [comment]
> Can you add in the PR why this actually fixes it?

Would be great to have that explanation in the commit message too :)

**Comment by rli** [comment]
> > Can you add in the PR why this actually fixes it?
> 
> Would be great to have that explanation in the commit message too :)

added!

**Pull request merged**